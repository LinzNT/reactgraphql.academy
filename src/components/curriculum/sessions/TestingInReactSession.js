import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

export const titleSession = 'Testing in React'

const TestingInReactSession = ({ title = titleSession }) => (
  <Session title={title}>
    <Ul>
      <Li>Unit Testing, Integration Testing, and Snapshot Testing</Li>
      <Li>Tooling: Enzyme & Jest</Li>
      <Li>Testing Components, Higher-Order Components, Connected Containers</Li>
      <Li>Unit tests Vs integration tests</Li>
      <Li>Code Coverage</Li>
    </Ul>
  </Session>
)

export default TestingInReactSession
