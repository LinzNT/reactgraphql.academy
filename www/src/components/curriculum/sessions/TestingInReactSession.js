import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

export const titleSession = 'Testing in React'

const TestingInReactSession = ({ title = titleSession }) => (
  <Session title={title}>
    <Ul>
      <Li>Unit Testing, Integration Testing, and Snapshot Testing</Li>
      <Li>Mocking and Dependency Injection in React </Li>
      <Li>Best practices to facilitate code refactoring</Li>
      <Li>Code Coverage</Li>
      <Li>
        Tooling:
        <Ul>
          <Li>React Testing Library</Li>
          <Li>Jest</Li>
          <Li>Enzyme</Li>
        </Ul>
      </Li>
    </Ul>
  </Session>
)

export default TestingInReactSession
