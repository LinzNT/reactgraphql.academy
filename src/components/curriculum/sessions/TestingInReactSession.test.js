import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import TestingInReactSession from './TestingInReactSession'

describe('<TestingInReactSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<TestingInReactSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
