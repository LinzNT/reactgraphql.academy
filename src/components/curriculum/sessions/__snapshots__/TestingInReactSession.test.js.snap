// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestingInReactSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      Unit Testing, Integration Testing, and Shapshot Testing
    </styled.li>
    <styled.li>
      Tooling: Enzyme & Jest
    </styled.li>
    <styled.li>
      Testing Components, Higher-Order Components, Connected Containers
    </styled.li>
    <styled.li>
      Unit tests Vs integration tests
    </styled.li>
    <styled.li>
      Code Coverage
    </styled.li>
  </styled.ul>
</Session>
`;
